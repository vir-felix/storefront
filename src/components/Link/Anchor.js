/* Anchor gets its content from the `children` prop */
/* eslint-disable jsx-a11y/anchor-has-content */
import React, { Component } from "react";
import PropTypes from "prop-types";
import track from "lib/tracking/track";

@track((ownProps) => ({
  component: "Anchor",
  url: ownProps.href
}))
export default class Anchor extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onAnchorClick: PropTypes.func,
    onClick: PropTypes.func
  }

  static defaultProps = {
    onAnchorClick: () => {},
    onClick: () => {}
  }

  @track(() => ({
    action: "Link Clicked"
  }))
  handleClick = (event) => {
    event.preventDefault();
    this.props.onClick(event);
    this.props.onAnchorClick(event);
  }

  @track(() => ({
    action: "Link Enter Key Down"
  }))
  handleKeyDown = (event) => {
    event.preventDefault();
    if (event.key === "Enter") {
      this.props.onClick(event);
      this.props.onAnchorClick(event);
    }
  }

  render() {
    const {
      onAnchorClick, // eslint-disable-line
      ...props
    } = this.props;

    return (
      <a
        {...props}
        onClick={this.handleClick}
        onKeyDown={this.handleKeyDown}
        role="link"
        tabIndex={0}
      />
    );
  }
}
