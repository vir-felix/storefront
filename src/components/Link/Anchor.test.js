import React from "react";
import renderer from "react-test-renderer";
import Anchor from "./Anchor";

test("basic snapshot", () => {
  const component = renderer.create(<Anchor href="/next">Next</Anchor>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
