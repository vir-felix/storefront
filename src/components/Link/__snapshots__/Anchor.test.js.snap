// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<a
  href="/next"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
  tracking={
    Object {
      "getTrackingData": [Function],
      "trackEvent": [Function],
    }
  }
>
  Next
</a>
`;
